src/command_modules/azure-cli-container/azure/cli/command_modules/container/custom.py

```diff
@@ -202,6 +214,85 @@ def create_container(cmd,
     return sdk_no_wait(no_wait, container_group_client.create_or_update, resource_group_name, name, cgroup)
 
 
+def _get_resource(client, resource_group_name, *subresources):
+    from msrestazure.azure_exceptions import CloudError
+    try:
+        resource = client.get(resource_group_name, *subresources)
+        return resource
+    except CloudError as ex:
+        if ex.error.error == "NotFound" or ex.error.error == "ResourceNotFound":
+            return None
+        else:
+            raise
+
+
+def _get_vnet_network_profile(cli_ctx, location, resource_group_name, vnet_name, vnet_address_prefix, subnet, subnet_address_prefix):
+    containerInstanceDelegationName = "Microsoft.ContainerInstance.containerGroups"
+
+    aci_delegation = Delegation(
+        name="Microsoft.ContainerInstance.containerGroups",
+        service_name="Microsoft.ContainerInstance/containerGroups"
+    )
+
+    ncf = cf_network(cli_ctx)
+
+    subnet_name = subnet
+    if is_valid_resource_id(subnet):
+        subnet_name = parse_resource_id(subnet)['resource_name']
```

Comment:
If subnet is a valid ID, you should probably also parse vnet name. In all other cases in the CLI, providing `--subnet ID --vnet-name NAME` raises an error.